cli/cmd/root.go

```diff
@@ -166,6 +168,32 @@ func Execute() (err error) {
 	return
 }
 
+func setupRootHelpCommand() {
+	// We want 'lacework help iac org list' to invoke 'iac help org list'
+	// instead of just showing help for lacework. The command in question is
+	// the default help command for the root command, so we peek at the
+	// target and if it's a component then invoke the component.
+	rootCmd.InitDefaultHelpCmd()
+	helpCommand, _, _ := rootCmd.Find([]string{"help"})
+	defaultRunHelp := helpCommand.Run
+	helpCommand.Run = nil
+	helpCommand.RunE = func(cmd *cobra.Command, args []string) error {
+		target, _, _ := rootCmd.Find(args)
+		if target != nil && isComponent(target.Annotations) {
+			f, ok := cli.LwComponents.GetComponent(target.Use)
+			if ok {
+				envs := []string{
+					fmt.Sprintf("LW_COMPONENT_NAME=%s", target.Use),
+				}
+				helpArgs := append([]string{"help"}, args[1:]...)
+				return f.RunAndOutput(helpArgs, envs...)
```

Comment:
This change it nice, my only concern is that if a component doesn't have a `help` command, then we will error out pretty nasty so, could we run the component but capture the `STDERR` and `STDOUT` before displaying it to the user and, if the command returns and exit code other than `0` then print something more helpful, maybe the `defaultRunHelp()` instead?

Here is an example of the output we would see for components that doesn't have a help command:
```
{"level":"error","ts":"2023-01-10T12:04:52-08:00","caller":"cmd/cdk_init.go:94","msg":"LW_ACCOUNT environment variable is not set","stacktrace":"... suppressed output ..."}
Usage:
  lacework help [command] [flags]

Flags:
  -h, --help   help for help

Global Flags:
  -a, --account string      account subdomain of URL (i.e. <ACCOUNT>.lacework.net)
  -k, --api_key string      access key id
  -s, --api_secret string   secret access key
      --api_token string    access token (replaces the use of api_key and api_secret)
      --debug               turn on debug logging
      --json                switch commands output from human-readable to json format
      --nocache             turn off caching
      --nocolor             turn off colors
      --noninteractive      turn off interactive mode (disable spinners, prompts, etc.)
      --organization        access organization level data sets (org admins only)
  -p, --profile string      switch between profiles configured at ~/.lacework.toml
      --subaccount string   sub-account name inside your organization (org admins only)

➜  go-sdk git:(component-help) ✗
```
Note that I suppressed the looooong stracktrace on purpose, but we all know how long this string will be 😓

Reply:
I just checked the existing components in dev and they all support a help command.

sca and sast don't work properly because they expect to always get a client.  The help command as-implemented doesn't give one, which I suppose it could.  Bit odd though to go to the trouble to get a client just to run help.

Reply:
Actually if you look closely, it's the component itself complaining that LW_ACCOUNT isn't set, not that it doesn't have a help command. 

I'm assuming you're looking at sca (or sast maybe, also assuming the code is similar.)  It does have a help command, it's just that it insists on getting an api client for the help command.  See https://github.com/lacework-dev/sca/blob/main/cmd/root.go#L66.

Reply:
Ill take a look at this early next week, most likely you are right 😄

Reply:
Thank you so much @slshen for looking into this. I completely agree with you, I will let @linpengzhang @edoardopirovano @declanwilson-lw know that, for help messages, we don't need to generate an API client.

I am +1 for this change! 💯